mistakedetection.tests/src/ca/mcgill/sel/mistakedetection/tests/MistakeDetectionPerformanceAnalysis.java

```diff
@@ -428,172 +307,218 @@ public static void produceExcelSheet(Comparison comparison, String name, String
-
-      Set<String> keyid = studentData.keySet();
-      int rowid = 0;
-
-      for (String key : keyid) {
-        XSSFRow row = spreadsheet.createRow(rowid++);
-        Object[] objectArr = studentData.get(key);
-        int cellid = 0;
-        for (var obj : objectArr) {
-          XSSFCell cell = row.createCell(cellid++);
-          if (obj instanceof String) {
-            String sObj = (String) obj;
-            if (sObj.startsWith("=")) {
-              sObj = sObj.replaceFirst("=", "");
-              cell.setCellFormula(sObj);
+  public static String getConcatNames(List<String> elems) {
+    return String.join(", ", elems);
+  }
+
+  /** Sets up and returns the hotel instructor solution. */
+  private static Solution setupHotelInstructorSolution() {
+    var instructorClassDiagram = cdmFromFile(HOTEL_INSTRUCTOR_SOLUTION);
+    var instructorSolution = instructorSolutionFromClassDiagram(instructorClassDiagram);
+    var tagGroup = setAbstractionTagToClassInClassDiag("RoomType", instructorClassDiagram, instructorSolution);
+    setOccurrenceTagToClassInClassDiag("Room", tagGroup, instructorClassDiagram);
+    setSynonymsFromFile(instructorSolution, HOTEL_SYNONYMS_FILE);
+    return instructorSolution;
+  }
+
+  /** Sets up and returns the hotel instructor solutions. */
+  private static List<Solution> setupSmartHomeInstructorSolutions() {
+    try (var files = Files.list(Path.of(SMART_HOME_INSTRUCTOR_SOLUTION_DIR))) {
+      return files.filter(f -> f.toString().endsWith(".cdm"))
+          .map(f -> setSynonymsFromFile(instructorSolutionFromClassDiagram(cdmFromFile(f)), SMART_HOME_SYNONYMS_FILE))
+          .collect(Collectors.toUnmodifiableList());
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+    return Collections.emptyList();
+  }
+
+  /* Sets synonyms in the given instructor solution based on the given properties file. */
+  private static Solution setSynonymsFromFile(Solution instSolution, String synonymsFile) {
+    var props = new Properties();
+    var instCdm = instSolution.getClassDiagram();
+    try (var fis = new FileInputStream(synonymsFile)) {
+      props.load(fis);
+      props.entrySet().forEach(e -> {
+        var name = (String) e.getKey();
+        var syns = Arrays.stream(((String) e.getValue()).split(",")).map(String::trim)
+            .collect(Collectors.toUnmodifiableList());
+        if (!name.contains(".")) {
+          try {
+            setSynonymToClassInClassDiag(name, syns, instCdm, instSolution);
+          } catch (Exception ex) { // ignored (some elements may not exist in all inst solution variations)
+          }
+        } else {
+          var splitName = name.split("\\.");
+          var className = splitName[0];
+          var propertyName = splitName[1];
+          try {
+            var cls = getClassFromClassDiagram(className, instCdm);
+            if (cls.getAttributes().stream().map(Attribute::getName).anyMatch(n -> n.equals(propertyName))) {
+              try {
```

Comment:
These ugly nested try-catches can be removed in the future if we find a better way to represent synonyms, eg, with a DSL